Frameworks/CoreLocation/CLLocationManager.mm

```diff
@@ -661,17 +751,47 @@ - (void)stopMonitoringSignificantLocationChanges {
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)startUpdatingHeading {
-    UNIMPLEMENTED();
+    @synchronized(self) {
+        if (!_periodicHeadingUpdateRequested) {
+            NSTraceInfo(TAG, @"Started periodic heading update");
+
+            if (self.allowsBackgroundLocationUpdates) {
+                // Request for a extended execution session so heading updates can continue in the background.
+                _requestExtendedExecutionSession();
+                _extendedExecutionSessionRequested = YES;
+            }
+
+            // Register for position change event only the first time heading update is requested.
+            __weak CLLocationManager* weakSelf = self;
+            _uwpPeriodicHeadingChangeToken =
+                [_uwpCompass addReadingChangedEvent:^void(WDSCompass* compass, WDSCompassReadingChangedEventArgs* event) {
+                    [weakSelf _handleHeadingChangedEvent:compass statusEvent:event];
+                }];
+            _periodicHeadingUpdateRequested = YES;
+        }
+    }
 }
 
 /**
- @Status Stub
+ @Status Interoperable
 */
 - (void)stopUpdatingHeading {
-    UNIMPLEMENTED();
+    @synchronized(self) {
+        if (_periodicHeadingUpdateRequested) {
+            NSTraceInfo(TAG, @"Stopped periodic heading update");
+            [_uwpCompass removeReadingChangedEvent:_uwpPeriodicHeadingChangeToken];
+            
+            if (_extendedExecutionSessionRequested) {
+                _removeExtendedExecutionSession();
+                _extendedExecutionSessionRequested = NO;
```

Comment:
This variable doesn't work properly; it's set/stomped in several spots.  It looks like _requestExtendedExecutionSession/_removeExtendedExecutionSession already does ref-counting, so you should be able to just call them as needed.  

The only place that needs special handling is dealloc - where we'll need to clear out the extended execution session regardless of s_extendedExecutionRequestCount's value.